### PR TITLE
Update GWTP Basic Archetype

### PR DESCRIPTION
**Pull request by jasonlemay**

### CODE REVIEW

File: archetypes/gwtp-basic/src/main/java/com/arcbees/gwtpbasic/client/application/home/HomeView.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.user.client.ui.Widget;
+import com.gwtplatform.mvp.client.ViewImpl;
+
+public class HomeView extends ViewImpl implements HomePresenter.MyView {
+    interface Binder extends UiBinder<Widget, HomeView> {
+    }
+
+    @Inject
+    public HomeView(
```
Comment (meriouma) [inline archetypes/gwtp-basic/src/main/java/com/arcbees/gwtpbasic/client/application/home/HomeView.java]: Remove public

File: archetypes/gwtp-basic/src/main/webapp/WEB-INF/appengine-web.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
+    <application>gwtp-sample-polymer</application>
```
Comment (meriouma) [inline archetypes/gwtp-basic/src/main/webapp/WEB-INF/appengine-web.xml]: gwtp-basic

File: archetypes/gwtp-basic/src/main/webapp/WEB-INF/web.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+<web-app xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://java.sun.com/xml/ns/javaee"
+         xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd"
+         version="2.5">
+
+    <!-- Servlets -->
+
+    <!-- Default page to serve -->
+    <welcome-file-list>
+        <welcome-file>gwtpsamplepolymer.html</welcome-file>
```
Comment (meriouma) [inline archetypes/gwtp-basic/src/main/webapp/WEB-INF/web.xml]: gwtpbasic.html?

File: archetypes/gwtp-basic/src/main/webapp/index.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<!doctype html>
+<html>
+<head>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0,maximum-scale=1.0">
+    <meta http-equiv="content-type" content="text/html; charset=UTF-8">
+    <title>GWTP - Boilerplate</title>
```
Comment (meriouma) [inline archetypes/gwtp-basic/src/main/webapp/index.html]: GWTP - Basic ?

**Comment by meriouma** [comment]
LGTM

File: archetypes/gwtp-basic/src/main/webapp/WEB-INF/web.xml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+<web-app xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://java.sun.com/xml/ns/javaee"
+         xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd"
+         version="2.5">
+
+    <!-- Servlets -->
+
+    <!-- Default page to serve -->
+    <welcome-file-list>
+        <welcome-file>gwtpbasic.html</welcome-file>
```
Comment (Chris-V) [inline archetypes/gwtp-basic/src/main/webapp/WEB-INF/web.xml]: index.html

File: archetypes/gwtp-basic/src/main/java/com/arcbees/gwtpbasic/GwtpBasic.gwt.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE module PUBLIC "-//Google Inc.//DTD Google Web Toolkit 2.7.0//EN"
+        "https://raw.githubusercontent.com/gwtproject/gwt/2.7.0/distro-source/core/src/gwt-module.dtd">
```
Comment (Chris-V) [inline archetypes/gwtp-basic/src/main/java/com/arcbees/gwtpbasic/GwtpBasic.gwt.xml]: http://gwtproject.org/doctype/2.7.0/gwt-module.dtd

File: archetypes/gwtp-basic/src/main/java/com/arcbees/gwtpbasic/client/application/home/HomeView.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+import com.gwtplatform.mvp.client.ViewImpl;
+
+public class HomeView extends ViewImpl implements HomePresenter.MyView {
+    interface Binder extends UiBinder<Widget, HomeView> {
+    }
+
+    @Inject
+    HomeView(
+            Binder uiBinder) {
+
```
Comment (Chris-V) [inline archetypes/gwtp-basic/src/main/java/com/arcbees/gwtpbasic/client/application/home/HomeView.java]: remove empty line. The ctor param can go go up as well

File: archetypes/gwtp-basic/pom.xml [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+                <artifactId>gwt-maven-plugin</artifactId>
+                <version>${gwt.version}</version>
+                <configuration>
+                    <strict>true</strict>
+
+                    <testTimeOut>180</testTimeOut>
+                    <mode>htmlunit</mode>
+                    <!-- Switch to a suite to run tests faster -->
+                    <!-- <includes>**/*GwtTestSuite.java</includes> -->
+                    <!-- <excludes>**/*GwtTest.java</excludes> -->
```
Comment (Chris-V) [inline archetypes/gwtp-basic/pom.xml]: I think we can get rid of these comments. We don't use gwt tests often enough

File: archetypes/gwtp-basic/pom.xml [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+
+    <!-- GWT (BOM) -->
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>com.google.gwt</groupId>
+                <artifactId>gwt</artifactId>
+                <version>${gwt.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
```
Comment (Chris-V) [inline archetypes/gwtp-basic/pom.xml]: Remove that. I'm not a fan of importing external dependency management

File: archetypes/gwtp-basic/pom.xml [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+
+    <dependencies>
+        <!-- GWT -->
+        <dependency>
+            <groupId>com.google.gwt</groupId>
+            <artifactId>gwt-user</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.google.gwt</groupId>
+            <artifactId>gwt-codeserver</artifactId>
```
Comment (Chris-V) [inline archetypes/gwtp-basic/pom.xml]: I think importing the codeserver is not needed anymore.

File: archetypes/gwtp-basic/pom.xml [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
+        <!-- GWT -->
+        <dependency>
+            <groupId>com.google.gwt</groupId>
+            <artifactId>gwt-user</artifactId>
+        </dependency>
```
Comment (Chris-V) [inline archetypes/gwtp-basic/pom.xml]: Add version and provided scope

File: archetypes/gwtp-basic/src/main/java/com/arcbees/gwtpbasic/GwtpBasic.gwt.xml (hunk 6 above)
Comment (branflake2267) [inline archetypes/gwtp-basic/src/main/java/com/arcbees/gwtpbasic/GwtpBasic.gwt.xml]: Just a thought when I see this. I think the response type on the server should be changed for these types: 
This is what we get, `Resource interpreted as Document but transferred with MIME type application/xml-dtd: "http://www.gwtproject.org/doctype/2.7.0/gwt-module.dtd".` Idea complains about this. :)

File: archetypes/gwtp-basic/pom.xml [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+            <artifactId>gwtp-dispatch-rest-shared</artifactId>
+            <version>${gwtp.version}</version>
+        </dependency>
+
+        <!-- DI -->
+        <dependency>
+            <groupId>com.google.gwt.inject</groupId>
+            <artifactId>gin</artifactId>
+            <version>${gin.version}</version>
+        </dependency>
```
Comment (Chris-V) [inline archetypes/gwtp-basic/pom.xml]: provided

**Comment by branflake2267** [comment]
Remind me when your done so I can regenerate the archetype.

**Comment by branflake2267** [comment]
Are these ready to be generated and deployed?

**Comment by meriouma** [comment]
Not yet, we're still working on it. In fact we're thinking of dropping the generator part and only keeping templates to allow using more variables. And we'll probably drop a few archetypes too, we want to keep them more simple than some are! But thanks anyway :)

**Comment by branflake2267** [comment]
Good idea, sounds good.

**Pull request merged**